src/opcode.rs

```diff
@@ -227,23 +227,36 @@ opcode! {
 }
 
 opcode! {
-    /// Read from pre-mapped buffers that have been previously registered with
+    /// Read from a file into a fixed buffer that has been previously registered with
     /// [`Submitter::register_buffers`](crate::Submitter::register_buffers).
     ///
+    /// The operation designated by this `opcode` will read data from a file at a specific offset
+    /// into a fixed buffer that has been registered into the `io-uring` instance via
+    /// [`Submitter::register_buffers`](crate::Submitter::register_buffers).
+    ///
+    /// The caller must ensure that the `buf_index` argument indexes into the list of buffers that
+    /// have been registered and that the buffer the index refers to has a base pointer of `buf` and
+    /// a length of `len`.
+    ///
     /// The return values match those documented in the `preadv2(2)` man pages.
+    ///
+    /// See the man pages for `io_uring_prep_read_fixed(3)` for more information.
     #[derive(Debug)]
     pub struct ReadFixed {
-        /// The `buf_index` is an index into an array of fixed buffers,
-        /// and is only valid if fixed buffers were registered.
+        /// This operation is only valid if `buf_index` is an index into an array of previously
+        /// registered fixed buffers, and the `buf` and `len` arguments fall within the region
+        /// specified by `buf_index`. Note that `buf` does not need to be aligned with the start of
```

Comment:
The added documentation seems unnecessary to me. Is this a behavior specific to ReadFixed?

Reply:
Not completely sure what you mean by behavior "specific" to `ReadFixed`, it is behavior specific to both `io_uring_prep_xxxxx_fixed`: https://man7.org/linux/man-pages/man3/io_uring_prep_read_fixed.3.html

> The buf and nbytes arguments mus fall within a region specified by buf_index in the previously registered buffer. The buffer need not be aligned with the start of the registered buffer.

And from the first io_uring [article](https://kernel.dk/io_uring.pdf):

> Upon successful registration of the buffers, the application can use the IORING_OP_READ_FIXED and IORING_OP_WRITE_FIXED to perform IO to and from these buffers. When using these fixed op-codes, sqe->addr must contain an address that is within one of these buffers, and sqe->len must contain the length (in bytes) of the request.